tripy/tripy/backend/compiler_api.py

```diff
@@ -31,6 +31,169 @@
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        out = compiled_linear(a)
+        tp.synchronize()
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+    default_stream().synchronize()
+
+
+@export.public_api(document_under="compiler")
+class Stream:
+    """
+    Represents a CUDA stream that can be used to manage concurrent operations.
+
+    This class is a wrapper around the underlying stream object, allowing management of CUDA streams.
+
+    Args:
+        priority (int): Assign priority for the new stream. Lower number signifies higher priority.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Creating new streams
+
+        streamA = tp.Stream()
+        streamB = tp.Stream()
+        defaultStream = tp.default_stream()
+        assert streamA != streamB
+        assert defaultStream != streamB
+
+    .. code-block:: python
+        :linenos:
+        :caption: Example
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        compiled_linear.stream = tp.Stream()
+        out = compiled_linear(a)
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+
+    def __init__(self, priority=0):
+        if priority != 0:
+            raise_error(
+                "Incorrect stream priority",
+                [f"Stream priority can only be 0 until #172 is fixed, got priority={priority}."],
+            )
+        from tripy.backend.mlir.utils import MLIRRuntimeClient
+
+        self._active_cuda_stream = MLIRRuntimeClient().create_stream()
+
+    def synchronize(self) -> None:
+        """Synchronize the stream, blocking until all operations in this stream are complete.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Use streams in execution
+
+            linear = tp.Linear(2, 3)
+            compiler = tp.Compiler(linear)
+
+            compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+            a = tp.ones((2, 2), dtype=tp.float32)
+            compiled_linear.stream = tp.Stream()
+            out = compiled_linear(a)
+            compiled_linear.stream.synchronize()
+            assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+        """
+        self._active_cuda_stream.sync()
+
+    def __eq__(self, other):
+        if not isinstance(other, Stream):
+            return False
+
+        if not (hasattr(self, "_active_cuda_stream") and hasattr(other, "_active_cuda_stream")):
+            return False
+
+        return self._active_cuda_stream == other._active_cuda_stream
+
+    def __repr__(self):
+        return f"<Stream(id={id(self)})>"
+
+    def __hash__(self):
+        return hash(id(self))
```

Comment:
Not sure you need any of these

Reply:
`__hash__` might not be required right now but I think its fine to keep it. `__str__` and `__eq__` are required.